### ISSUE TITLE
Internal error initializing Criteo instance (version 4.8.0)

### ISSUE THREAD

**Issue by manyasone** [criteo/android-publisher-sdk#270]
Hi,

I'm not using the android-publisher-sdk directly - it is a sub dependency of the yield love sdk which I am using.

Since today I'm getting an error for the latest version 4.8.0:
```
 com.criteo.publisher.CriteoInitException: Internal error initializing Criteo instance.
                                                                                                    	at com.criteo.publisher.Criteo.init(Criteo.java:146)
                                                                                                    	at com.criteo.publisher.Criteo.access$000(Criteo.java:45)
                                                                                                    	at com.criteo.publisher.Criteo$Builder.init(Criteo.java:109)
                                                                                                    	at com.yieldlove.adIntegration.SdkAdapters.criteo.CriteoSdkAdapter.initCriteoSdk(CriteoSdkAdapter.java:156)
...
```

When switching back to 4.7.0, everything seems to be working fine, so there seems to be an issue with your latest release.

**Comment by viktor-rasevych-criteo** [comment]
Hi @manyasone ,

Could you share full stacktrace to make sure that i reproduced the same issue?

Thanks, Viktor

**Comment by manyasone** [comment]
Hi @viktor-rasevych-criteo, 

sure, here's the full stacktrace (pardon the strange new logcat formatting...).

```
--------- beginning of crash
2022-10-05 10:06:12.871  5872-5872  AndroidRuntime          de.msg                               E  FATAL EXCEPTION: main
                                                                                                    Process: de.msg, PID: 5872
                                                                                                    java.lang.AssertionError: Missing field in com.criteo.publisher.logging.RemoteLogRecords$a
                                                                                                    	at com.squareup.moshi.adapters.EnumJsonAdapter.<init>(EnumJsonAdapter.java:75)
                                                                                                    	at com.squareup.moshi.adapters.EnumJsonAdapter.create(EnumJsonAdapter.java:49)
                                                                                                    	at com.criteo.publisher.s.I(DependencyProvider.java:617)
                                                                                                    	at com.criteo.publisher.s.$r8$lambda$1zlcPCZA5uuZ9cDkcVIv7_0KxKg(Unknown Source:0)
                                                                                                    	at com.criteo.publisher.s$$ExternalSyntheticLambda41.a(Unknown Source:0)
                                                                                                    	at com.criteo.publisher.s$$ExternalSyntheticLambda54.invoke(Unknown Source:2)
                                                                                                    	at com.criteo.publisher.m0.k.a(MapUtil.kt:38)
                                                                                                    	at com.criteo.publisher.s.a(DependencyProvider.java:511)
                                                                                                    	at com.criteo.publisher.s.O0(DependencyProvider.java:614)
                                                                                                    	at com.criteo.publisher.s.B(DependencyProvider.java:608)
                                                                                                    	at com.criteo.publisher.s.$r8$lambda$1DrYamo_fc1Jaqw4lqprW1mJecg(Unknown Source:0)
                                                                                                    	at com.criteo.publisher.s$$ExternalSyntheticLambda0.a(Unknown Source:2)
                                                                                                    	at com.criteo.publisher.s$$ExternalSyntheticLambda54.invoke(Unknown Source:2)
                                                                                                    	at com.criteo.publisher.m0.k.a(MapUtil.kt:38)
                                                                                                    	at com.criteo.publisher.s.a(DependencyProvider.java:511)
                                                                                                    	at com.criteo.publisher.s.G0(DependencyProvider.java:607)
                                                                                                    	at com.criteo.publisher.s.a(DependencyProvider.java:582)
                                                                                                    	at com.criteo.publisher.s.Q(DependencyProvider.java:677)
                                                                                                    	at com.criteo.publisher.s.$r8$lambda$HT0UurXJiC9cj8jUeK87uKDXYR4(Unknown Source:0)
                                                                                                    	at com.criteo.publisher.s$$ExternalSyntheticLambda2.a(Unknown Source:2)
                                                                                                    	at com.criteo.publisher.s$$ExternalSyntheticLambda54.invoke(Unknown Source:2)
                                                                                                    	at com.criteo.publisher.m0.k.a(MapUtil.kt:38)
                                                                                                    	at com.criteo.publisher.s.a(DependencyProvider.java:511)
                                                                                                    	at com.criteo.publisher.s.X0(DependencyProvider.java:676)
                                                                                                    	at com.criteo.publisher.s.O(DependencyProvider.java:710)
                                                                                                    	at com.criteo.publisher.s.$r8$lambda$gTxppZSH7f0KUZPeYrCBWYT6DXE(Unknown Source:0)
                                                                                                    	at com.criteo.publisher.s$$ExternalSyntheticLambda21.a(Unknown Source:2)
                                                                                                    	at com.criteo.publisher.s$$ExternalSyntheticLambda54.invoke(Unknown Source:2)
                                                                                                    	at com.criteo.publisher.m0.k.a(MapUtil.kt:38)
                                                                                                    	at com.criteo.publisher.s.a(DependencyProvider.java:511)
                                                                                                    	at com.criteo.publisher.s.V0(DependencyProvider.java:708)
                                                                                                    	at com.criteo.publisher.s$$ExternalSyntheticLambda56.invoke(Unknown Source:2)
                                                                                                    	at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
                                                                                                    	at com.criteo.publisher.f0.a.b(LazyDependency.kt:46)
                                                                                                    	at com.criteo.publisher.f0.a.a(LazyDependency.kt:48)
                                                                                                    	at com.criteo.publisher.logging.g.a(Logger.java:112)
                                                                                                    	at com.criteo.publisher.Criteo.init(Criteo.java:150)
                                                                                                    	at com.criteo.publisher.Criteo.access$000(Criteo.java:45)
                                                                                                    	at com.criteo.publisher.Criteo$Builder.init(Criteo.java:109)
                                                                                                    	at com.yieldlove.adIntegration.SdkAdapters.criteo.CriteoSdkAdapter.initCriteoSdk(CriteoSdkAdapter.java:156)
                                                                                                    	at com.yieldlove.adIntegration.SdkAdapters.criteo.CriteoSdkAdapter.$r8$lambda$ypjo_jkPGxBFn66O55kbMxCQELQ(Unknown Source:0)
                                                                                                    	at com.yieldlove.adIntegration.SdkAdapters.criteo.CriteoSdkAdapter$$ExternalSyntheticLambda4.run(Unknown Source:4)
                                                                                                    	at com.yieldlove.androidpromise.Promise.lambda$then$10(Promise.java:133)
                                                                                                    	at com.yieldlove.androidpromise.Promise$$ExternalSyntheticLambda1.run(Unknown Source:2)
                                                                                                    	at com.yieldlove.androidpromise.Promise.run(Promise.java:188)
                                                                                                    	at com.yieldlove.androidpromise.Promise.invokeThen(Promise.java:220)
                                                                                                    	at com.yieldlove.androidpromise.Promise.continueInChain(Promise.java:204)
                                                                                                    	at com.yieldlove.androidpromise.Promise.then(Promise.java:125)
                                                                                                    	at com.yieldlove.androidpromise.Promise.then(Promise.java:131)
                                                                                                    	at com.yieldlove.adIntegration.SdkAdapters.criteo.CriteoSdkAdapter.prepare(CriteoSdkAdapter.java:145)
                                                                                                    	at com.yieldlove.adIntegration.SdkAdapters.SdkManager.getBidFromAdapter(SdkManager.java:40)
                                                                                                    	at com.yieldlove.adIntegration.SdkAdapters.SdkManager.$r8$lambda$tdMongrSbGQh02u4nm5ncJoKVDU(Unknown Source:0)
                                                                                                    	at com.yieldlove.adIntegration.SdkAdapters.SdkManager$$ExternalSyntheticLambda1.run(Unknown Source:4)
                                                                                                    	at com.yieldlove.androidpromise.Promise.map(Promise.java:44)
2022-10-05 10:06:12.871  5872-5872  AndroidRuntime          de.msg                               E  	at com.yieldlove.adIntegration.SdkAdapters.SdkManager.getBids(SdkManager.java:28)
                                                                                                    	at com.yieldlove.adIntegration.AdFormats.YieldloveBannerAd.getBid(YieldloveBannerAd.java:159)
                                                                                                    	at com.yieldlove.adIntegration.AdFormats.YieldloveBannerAd.$r8$lambda$nqVqQRd9-sQj6ZArtFxExrPcBpg(Unknown Source:0)
                                                                                                    	at com.yieldlove.adIntegration.AdFormats.YieldloveBannerAd$$ExternalSyntheticLambda5.run(Unknown Source:2)
                                                                                                    	at com.yieldlove.androidpromise.Promise.lambda$then$7(Promise.java:97)
                                                                                                    	at com.yieldlove.androidpromise.Promise$$ExternalSyntheticLambda6.run(Unknown Source:2)
                                                                                                    	at com.yieldlove.androidpromise.Promise.lambda$then$8(Promise.java:103)
                                                                                                    	at com.yieldlove.androidpromise.Promise$$ExternalSyntheticLambda11.run(Unknown Source:2)
                                                                                                    	at com.yieldlove.androidpromise.Promise.run(Promise.java:188)
                                                                                                    	at com.yieldlove.androidpromise.Promise.invokeThen(Promise.java:220)
                                                                                                    	at com.yieldlove.androidpromise.Promise.continueInChain(Promise.java:204)
                                                                                                    	at com.yieldlove.androidpromise.Promise.then(Promise.java:125)
                                                                                                    	at com.yieldlove.androidpromise.Promise.then(Promise.java:101)
                                                                                                    	at com.yieldlove.androidpromise.Promise.then(Promise.java:97)
                                                                                                    	at com.yieldlove.adIntegration.AdFormats.YieldloveBannerAd.load(YieldloveBannerAd.java:85)
                                                                                                    	at app.ads.GoogleListAdLoader.initAdView(GoogleListAdLoader.kt:103)
                                                                                                    	at app.ads.GoogleListAdLoader.initViews(GoogleListAdLoader.kt:70)
                                                                                                    	at app.ads.GoogleListAdLoader.refreshAd(GoogleListAdLoader.kt:56)
                                                                                                    	at app.activities.main.list.SearchResultsAdapter.onCreateViewHolder(SearchResultsAdapter.kt:158)
                                                                                                    	at androidx.recyclerview.widget.RecyclerView$Adapter.createViewHolder(RecyclerView.java:7295)
                                                                                                    	at androidx.recyclerview.widget.RecyclerView$Recycler.tryGetViewHolderForPositionByDeadline(RecyclerView.java:6416)
                                                                                                    	at androidx.recyclerview.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:6300)
                                                                                                    	at androidx.recyclerview.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:6296)
                                                                                                    	at androidx.recyclerview.widget.LinearLayoutManager$LayoutState.next(LinearLayoutManager.java:2330)
                                                                                                    	at androidx.recyclerview.widget.LinearLayoutManager.layoutChunk(LinearLayoutManager.java:1631)
                                                                                                    	at androidx.recyclerview.widget.LinearLayoutManager.fill(LinearLayoutManager.java:1591)
                                                                                                    	at androidx.recyclerview.widget.LinearLayoutManager.onLayoutChildren(LinearLayoutManager.java:668)
                                                                                                    	at androidx.recyclerview.widget.RecyclerView.dispatchLayoutStep2(RecyclerView.java:4309)
                                                                                                    	at androidx.recyclerview.widget.RecyclerView.dispatchLayout(RecyclerView.java:4012)
                                                                                                    	at androidx.recyclerview.widget.RecyclerView.onLayout(RecyclerView.java:4578)
                                                                                                    	at android.view.View.layout(View.java:23143)
                                                                                                    	at android.view.ViewGroup.layout(ViewGroup.java:6412)
                                                                                                    	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1829)
                                                                                                    	at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1673)
                                                                                                    	at android.widget.LinearLayout.onLayout(LinearLayout.java:1582)
                                                                                                    	at android.view.View.layout(View.java:23143)
                                                                                                    	at android.view.ViewGroup.layout(ViewGroup.java:6412)
                                                                                                    	at android.widget.RelativeLayout.onLayout(RelativeLayout.java:1103)
                                                                                                    	at android.view.View.layout(View.java:23143)
                                                                                                    	at android.view.ViewGroup.layout(ViewGroup.java:6412)
                                                                                                    	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1829)
                                                                                                    	at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1673)
                                                                                                    	at android.widget.LinearLayout.onLayout(LinearLayout.java:1582)
                                                                                                    	at android.view.View.layout(View.java:23143)
                                                                                                    	at android.view.ViewGroup.layout(ViewGroup.java:6412)
                                                                                                    	at androidx.coordinatorlayout.widget.CoordinatorLayout.layoutChild(CoordinatorLayout.java:1213)
                                                                                                    	at androidx.coordinatorlayout.widget.CoordinatorLayout.onLayoutChild(CoordinatorLayout.java:899)
                                                                                                    	at com.google.android.material.bottomsheet.BottomSheetBehavior.onLayoutChild(BottomSheetBehavior.java:537)
                                                                                                    	at androidx.coordinatorlayout.widget.CoordinatorLayout.onLayout(CoordinatorLayout.java:918)
                                                                                                    	at android.view.View.layout(View.java:23143)
                                                                                                    	at android.view.ViewGroup.layout(ViewGroup.java:6412)
                                                                                                    	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
2022-10-05 10:06:12.872  5872-5872  AndroidRuntime          de.msg                               E  	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
                                                                                                    	at android.view.View.layout(View.java:23143)
                                                                                                    	at android.view.ViewGroup.layout(ViewGroup.java:6412)
                                                                                                    	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
                                                                                                    	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
                                                                                                    	at android.view.View.layout(View.java:23143)
                                                                                                    	at android.view.ViewGroup.layout(ViewGroup.java:6412)
                                                                                                    	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1829)
                                                                                                    	at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1673)
                                                                                                    	at android.widget.LinearLayout.onLayout(LinearLayout.java:1582)
                                                                                                    	at android.view.View.layout(View.java:23143)
                                                                                                    	at android.view.ViewGroup.layout(ViewGroup.java:6412)
                                                                                                    	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
                                                                                                    	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
                                                                                                    	at android.view.View.layout(View.java:23143)
                                                                                                    	at android.view.ViewGroup.layout(ViewGroup.java:6412)
                                                                                                    	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1829)
                                                                                                    	at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1673)
                                                                                                    	at android.widget.LinearLayout.onLayout(LinearLayout.java:1582)
                                                                                                    	at android.view.View.layout(View.java:23143)
                                                                                                    	at android.view.ViewGroup.layout(ViewGroup.java:6412)
                                                                                                    	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
                                                                                                    	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
                                                                                                    	at com.android.internal.policy.DecorView.onLayout(DecorView.java:798)
                                                                                                    	at android.view.View.layout(View.java:23143)
                                                                                                    	at android.view.ViewGroup.layout(ViewGroup.java:6412)
                                                                                                    	at android.view.ViewRootImpl.performLayout(ViewRootImpl.java:3684)
                                                                                                    	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:3143)
                                                                                                    	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:2126)
                                                                                                    	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:8653)
                                                                                                    	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1037)
                                                                                                    	at android.view.Choreographer.doCallbacks(Choreographer.java:845)
                                                                                                    	at android.view.Choreographer.doFrame(Choreographer.java:780)
                                                                                                    	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1022)
                                                                                                    	at android.os.Handler.handleCallback(Handler.java:938)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:99)
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:201)
                                                                                                    	at android.os.Looper.loop(Looper.java:288)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:7839)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
                                                                                                    Caused by: java.lang.NoSuchFieldException: DEBUG
                                                                                                    	at java.lang.Class.getField(Class.java:1635)
                                                                                                    	at com.squareup.moshi.adapters.EnumJsonAdapter.<init>(EnumJsonAdapter.java:71)
                                                                                                    	... 147 more
```

**Comment by ondra-kormanik** [comment]
Hi, I can confirm the problem. Here are logs from our testing app
```
2022-10-05 10:18:09.028 1857-1857/com.yieldlove.exampleapp E/CriteoSdkCriteo: com.criteo.publisher.CriteoInitException: Internal error initializing Criteo instance.
        at com.criteo.publisher.Criteo.init(Criteo.java:146)
        at com.criteo.publisher.Criteo.access$000(Criteo.java:45)
        at com.criteo.publisher.Criteo$Builder.init(Criteo.java:109)
        at com.yieldlove.adIntegration.SdkAdapters.criteo.CriteoWrapper.init(CriteoWrapper.java:27)
        at com.yieldlove.adIntegration.SdkAdapters.criteo.CriteoSdkAdapter.initCriteoSdk(CriteoSdkAdapter.java:109)
        at com.yieldlove.adIntegration.SdkAdapters.criteo.CriteoSdkAdapter.$r8$lambda$ypjo_jkPGxBFn66O55kbMxCQELQ(Unknown Source:0)
        at com.yieldlove.adIntegration.SdkAdapters.criteo.CriteoSdkAdapter$$ExternalSyntheticLambda2.run(Unknown Source:4)
        at com.yieldlove.androidpromise.Promise.lambda$then$10(Promise.java:133)
        at com.yieldlove.androidpromise.Promise$$ExternalSyntheticLambda1.run(Unknown Source:2)
        at com.yieldlove.androidpromise.Promise.run(Promise.java:188)
        at com.yieldlove.androidpromise.Promise.invokeThen(Promise.java:220)
        at com.yieldlove.androidpromise.Promise.continueInChain(Promise.java:204)
        at com.yieldlove.androidpromise.Promise.then(Promise.java:125)
        at com.yieldlove.androidpromise.Promise.then(Promise.java:131)
        at com.yieldlove.adIntegration.SdkAdapters.criteo.CriteoSdkAdapter.prepare(CriteoSdkAdapter.java:101)
        at com.yieldlove.adIntegration.SdkAdapters.SdkManager.getBidFromAdapter(SdkManager.java:50)
        at com.yieldlove.adIntegration.SdkAdapters.SdkManager.$r8$lambda$tdMongrSbGQh02u4nm5ncJoKVDU(Unknown Source:0)
        at com.yieldlove.adIntegration.SdkAdapters.SdkManager$$ExternalSyntheticLambda1.run(Unknown Source:4)
        at com.yieldlove.androidpromise.Promise.map(Promise.java:44)
        at com.yieldlove.adIntegration.SdkAdapters.SdkManager.getBids(SdkManager.java:36)
        at com.yieldlove.adIntegration.AdFormats.YieldloveAd.getBid(YieldloveAd.java:110)
        at com.yieldlove.adIntegration.AdFormats.YieldloveBannerAd$$ExternalSyntheticLambda10.run(Unknown Source:2)
        at com.yieldlove.androidpromise.Promise.lambda$then$7(Promise.java:97)
        at com.yieldlove.androidpromise.Promise$$ExternalSyntheticLambda6.run(Unknown Source:2)
        at com.yieldlove.androidpromise.Promise.lambda$then$8(Promise.java:103)
        at com.yieldlove.androidpromise.Promise$$ExternalSyntheticLambda11.run(Unknown Source:2)
        at com.yieldlove.androidpromise.Promise.run(Promise.java:188)
        at com.yieldlove.androidpromise.Promise.invokeThen(Promise.java:220)
        at com.yieldlove.androidpromise.Promise.continueInChain(Promise.java:204)
        at com.yieldlove.androidpromise.Promise.then(Promise.java:125)
        at com.yieldlove.androidpromise.Promise.then(Promise.java:101)
        at com.yieldlove.androidpromise.Promise.then(Promise.java:97)
        at com.yieldlove.adIntegration.AdFormats.YieldloveBannerAd.load(YieldloveBannerAd.java:94)
        at com.yieldlove.exampleapp.MainActivity.loadBannerUsingExternalConfiguration(MainActivity.java:188)
        at com.yieldlove.exampleapp.MainActivity.findAdContainerAndLoadBanner(MainActivity.java:73)
        at com.yieldlove.exampleapp.MainActivity.showBanner(MainActivity.java:59)
        at com.yieldlove.exampleapp.MainActivity.onCreate(MainActivity.java:48)
        at android.app.Activity.performCreate(Activity.java:8068)
        at android.app.Activity.performCreate(Activity.java:8048)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1341)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3627)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3803)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:104)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2217)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loopOnce(Looper.java:201)
        at android.os.Looper.loop(Looper.java:288)
2022-10-05 10:18:09.028 1857-1857/com.yieldlove.exampleapp E/CriteoSdkCriteo:     at android.app.ActivityThread.main(ActivityThread.java:7813)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
     Caused by: java.lang.AssertionError: Missing field in com.criteo.publisher.logging.RemoteLogRecords$a
        at com.squareup.moshi.adapters.EnumJsonAdapter.<init>(EnumJsonAdapter.java:76)
        at com.squareup.moshi.adapters.EnumJsonAdapter.create(EnumJsonAdapter.java:48)
        at com.criteo.publisher.s.I(DependencyProvider.java:617)
        at com.criteo.publisher.s.$r8$lambda$1zlcPCZA5uuZ9cDkcVIv7_0KxKg(Unknown Source:0)
        at com.criteo.publisher.s$$ExternalSyntheticLambda41.a(Unknown Source:0)
        at com.criteo.publisher.s$$ExternalSyntheticLambda54.invoke(Unknown Source:2)
        at com.criteo.publisher.m0.k.a(MapUtil.kt:38)
        at com.criteo.publisher.s.a(DependencyProvider.java:511)
        at com.criteo.publisher.s.O0(DependencyProvider.java:614)
        at com.criteo.publisher.s.B(DependencyProvider.java:608)
        at com.criteo.publisher.s.$r8$lambda$1DrYamo_fc1Jaqw4lqprW1mJecg(Unknown Source:0)
        at com.criteo.publisher.s$$ExternalSyntheticLambda0.a(Unknown Source:2)
        at com.criteo.publisher.s$$ExternalSyntheticLambda54.invoke(Unknown Source:2)
        at com.criteo.publisher.m0.k.a(MapUtil.kt:38)
        at com.criteo.publisher.s.a(DependencyProvider.java:511)
        at com.criteo.publisher.s.G0(DependencyProvider.java:607)
        at com.criteo.publisher.s.o(DependencyProvider.java:254)
        at com.criteo.publisher.s.$r8$lambda$kg-j1-XQoNam_N90M5GOIHgeycY(Unknown Source:0)
        at com.criteo.publisher.s$$ExternalSyntheticLambda27.a(Unknown Source:2)
        at com.criteo.publisher.s$$ExternalSyntheticLambda54.invoke(Unknown Source:2)
        at com.criteo.publisher.m0.k.a(MapUtil.kt:38)
        at com.criteo.publisher.s.a(DependencyProvider.java:511)
        at com.criteo.publisher.s.q0(DependencyProvider.java:252)
        at com.criteo.publisher.m.<init>(CriteoInternal.java:83)
        at com.criteo.publisher.Criteo.init(Criteo.java:133)
        	... 52 more
     Caused by: java.lang.NoSuchFieldException: DEBUG
        at java.lang.Class.getField(Class.java:1635)
        at com.squareup.moshi.adapters.EnumJsonAdapter.<init>(EnumJsonAdapter.java:70)
        	... 76 more
2022-10-05 10:18:09.048 1857-1857/com.yieldlove.exampleapp E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.yieldlove.exampleapp, PID: 1857
    java.lang.AssertionError: Missing field in com.criteo.publisher.logging.RemoteLogRecords$a
        at com.squareup.moshi.adapters.EnumJsonAdapter.<init>(EnumJsonAdapter.java:76)
        at com.squareup.moshi.adapters.EnumJsonAdapter.create(EnumJsonAdapter.java:48)
        at com.criteo.publisher.s.I(DependencyProvider.java:617)
        at com.criteo.publisher.s.$r8$lambda$1zlcPCZA5uuZ9cDkcVIv7_0KxKg(Unknown Source:0)
        at com.criteo.publisher.s$$ExternalSyntheticLambda41.a(Unknown Source:0)
        at com.criteo.publisher.s$$ExternalSyntheticLambda54.invoke(Unknown Source:2)
        at com.criteo.publisher.m0.k.a(MapUtil.kt:38)
        at com.criteo.publisher.s.a(DependencyProvider.java:511)
        at com.criteo.publisher.s.O0(DependencyProvider.java:614)
        at com.criteo.publisher.s.B(DependencyProvider.java:608)
        at com.criteo.publisher.s.$r8$lambda$1DrYamo_fc1Jaqw4lqprW1mJecg(Unknown Source:0)
        at com.criteo.publisher.s$$ExternalSyntheticLambda0.a(Unknown Source:2)
        at com.criteo.publisher.s$$ExternalSyntheticLambda54.invoke(Unknown Source:2)
        at com.criteo.publisher.m0.k.a(MapUtil.kt:38)
        at com.criteo.publisher.s.a(DependencyProvider.java:511)
        at com.criteo.publisher.s.G0(DependencyProvider.java:607)
        at com.criteo.publisher.s.a(DependencyProvider.java:582)
        at com.criteo.publisher.s.Q(DependencyProvider.java:677)
        at com.criteo.publisher.s.$r8$lambda$HT0UurXJiC9cj8jUeK87uKDXYR4(Unknown Source:0)
        at com.criteo.publisher.s$$ExternalSyntheticLambda2.a(Unknown Source:2)
        at com.criteo.publisher.s$$ExternalSyntheticLambda54.invoke(Unknown Source:2)
        at com.criteo.publisher.m0.k.a(MapUtil.kt:38)
        at com.criteo.publisher.s.a(DependencyProvider.java:511)
        at com.criteo.publisher.s.X0(DependencyProvider.java:676)
        at com.criteo.publisher.s.O(DependencyProvider.java:710)
        at com.criteo.publisher.s.$r8$lambda$gTxppZSH7f0KUZPeYrCBWYT6DXE(Unknown Source:0)
        at com.criteo.publisher.s$$ExternalSyntheticLambda21.a(Unknown Source:2)
        at com.criteo.publisher.s$$ExternalSyntheticLambda54.invoke(Unknown Source:2)
        at com.criteo.publisher.m0.k.a(MapUtil.kt:38)
        at com.criteo.publisher.s.a(DependencyProvider.java:511)
        at com.criteo.publisher.s.V0(DependencyProvider.java:708)
        at com.criteo.publisher.s$$ExternalSyntheticLambda56.invoke(Unknown Source:2)
        at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
        at com.criteo.publisher.f0.a.b(LazyDependency.kt:46)
        at com.criteo.publisher.f0.a.a(LazyDependency.kt:48)
        at com.criteo.publisher.logging.g.a(Logger.java:112)
        at com.criteo.publisher.Criteo.init(Criteo.java:150)
        at com.criteo.publisher.Criteo.access$000(Criteo.java:45)
        at com.criteo.publisher.Criteo$Builder.init(Criteo.java:109)
        at com.yieldlove.adIntegration.SdkAdapters.criteo.CriteoWrapper.init(CriteoWrapper.java:27)
        at com.yieldlove.adIntegration.SdkAdapters.criteo.CriteoSdkAdapter.initCriteoSdk(CriteoSdkAdapter.java:109)
        at com.yieldlove.adIntegration.SdkAdapters.criteo.CriteoSdkAdapter.$r8$lambda$ypjo_jkPGxBFn66O55kbMxCQELQ(Unknown Source:0)
        at com.yieldlove.adIntegration.SdkAdapters.criteo.CriteoSdkAdapter$$ExternalSyntheticLambda2.run(Unknown Source:4)
        at com.yieldlove.androidpromise.Promise.lambda$then$10(Promise.java:133)
        at com.yieldlove.androidpromise.Promise$$ExternalSyntheticLambda1.run(Unknown Source:2)
        at com.yieldlove.androidpromise.Promise.run(Promise.java:188)
        at com.yieldlove.androidpromise.Promise.invokeThen(Promise.java:220)
        at com.yieldlove.androidpromise.Promise.continueInChain(Promise.java:204)
        at com.yieldlove.androidpromise.Promise.then(Promise.java:125)
        at com.yieldlove.androidpromise.Promise.then(Promise.java:131)
        at com.yieldlove.adIntegration.SdkAdapters.criteo.CriteoSdkAdapter.prepare(CriteoSdkAdapter.java:101)
        at com.yieldlove.adIntegration.SdkAdapters.SdkManager.getBidFromAdapter(SdkManager.java:50)
        at com.yieldlove.adIntegration.SdkAdapters.SdkManager.$r8$lambda$tdMongrSbGQh02u4nm5ncJoKVDU(Unknown Source:0)
```
Something goes wrong after calling Criteo `init` in our code from version 4.8.0, version 4.7.0 works fine for us.
(If there is someone using Yieldlove SDK, use v6.15.3 please. It is using concrete version 4.7.0.)

**Comment by viktor-rasevych-criteo** [comment]
Thanks for the stacktrance, already found the problem, will be releasing new version soon

**Comment by ondra-kormanik** [comment]
Hi @viktor-rasevych-criteo, v4.8.1 looks good for us, thank you. 👍

**Comment by viktor-rasevych-criteo** [comment]
Closing the issue because the fix was released in v4.8.1.
Feel free to reopen or create a new issue if it occurs again.
